### PR TITLE
Refine handle agg and exchange sender

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #4118 

Problem Summary:

### What is changed and how it works?

- add `AggregationInterpreterHelper` and `ExchangeSenderInterpreterHelper` to refine the code of `executeAgg` and `handleExchangeSender`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/AggregationInterpreterHelper.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        keys.push_back(before_agg_header.getPositionByName(name));
+    }
+
+    const Settings & settings = context.getSettingsRef();
+
+    /** Two-level aggregation is useful in two cases:
+      * 1. Parallel aggregation is done, and the results should be merged in parallel.
+      * 2. An aggregation is done with store of temporary data on the disk, and they need to be merged in a memory efficient way.
+      */
+    bool allow_to_use_two_level_group_by = before_agg_streams_size > 1 || settings.max_bytes_before_external_group_by != 0;
```
Comment (ywqzzy) [inline dbms/src/Flash/Coprocessor/AggregationInterpreterHelper.cpp:80]: nit: maybe extract the `two_level` logic in a function is cleaner.

Comment (SeaRise) [reply dbms/src/Flash/Coprocessor/AggregationInterpreterHelper.cpp]: done

**Review by windtalker — APPROVED** [review]
LGTM

**Review by ywqzzy — APPROVED** [review]
LGTM

**Pull request merged**